### DISCUSSION TITLE
New development - roadmap?

### DISCUSSION THREAD

**Discussion by richb-hanover** [sqm-autorate/sqm-autorate#169] [General]
@Fail-Safe Wow! That's a lot of activity! Congratulations!

What's the plan for enhancements to this code base? Thanks!

**Comment by Fail-Safe** [comment]
No significant plans at the moment. I'm testing @chas-iot's `delay-histogram.lua` plugin right now and I'm liking it a lot. Giving it some time to bake and see how it performs under varying conditions for several days.

**Comment by richb-hanover** [comment]
I submitted a wacky idea over on the bash implementation [repo](https://github.com/lynxthecat/cake-autorate/issues/115) but I'll mention it here, too. The gist was that sqm-autorate could record/log the current link rate at the onset of high latency, _just before_ throttling back the SQM settings. 

Why? I have 75/75 mbps symmetrical fiber service. But I suspect my ISP may bog down from time to time (likely sagging in the evening hours). But I don't know for sure, and don't want to spend my evening hours running speed tests.

But sqm-autorate is always running measuring link speeds. If it could log those peak speeds:

1. We would know the max speeds over time, without requiring regular speed tests. (We'd simply be using the data that sqm-autorate was already generating.) 

2. We would know how rapidly the link speed varies. Some links get slow in the evening (cable modems, my ISP?). Other links might change from minute to minute, or even second to second. It would be interesting to know this.
 
3. If the link speed varies more slowly, it might give an opportunity to slow the pinging of the reflectors, decreasing CPU load and traffic. 

Does this make sense? Thanks.
